src/api-next/api-next.ts

```diff
@@ -1658,6 +1657,74 @@ export class IBApiNext {
     );
   }
 
+  /** nextValidId event handler */
+  private readonly onNextValidId = (
+    subscriptions: Map<number, IBApiNextSubscription<number>>,
+    orderId: number
+  ): void => {
+    // this is special to other one-shot callbacks:
+    // we only want to complete one subscription at a time,
+    // to avoid multiple getNextValidOrderId calls to return same value
+    const next = subscriptions.entries().next();
+    if (next && !next.done && next.value[1]) {
+      next.value[1].next({
+        all: orderId,
+      });
+      next.value[1].complete();
+    }
+  };
+
+  /**
+   * Requests the next valid order ID at the current moment.
+   */
+  getNextValidOrderId(): Promise<number> {
+    return lastValueFrom(
+      this.subscriptions
+        .register<number>(
+          () => {
+            this.api.reqIds();
+          },
+          undefined,
+          [[EventName.nextValidId, this.onNextValidId]]
+        )
+        .pipe(map((v: { all: number }) => v.all))
+    );
+  }
```

Comment:
There is probably something wrong, maybe we don't end the sub listening?

There is an issue with timing out rightnow.